pyrsa/data/dataset.py

```diff
@@ -299,7 +299,37 @@ def sort_by(self, by):
         order = np.argsort(desc)
         self.measurements = self.measurements[order]
         self.obs_descriptors = subset_descriptor(self.obs_descriptors, order)
+        
+    def get_measurements(self):
+        "Getter function for measurements"
+        return self.measurements.copy()
+    
+    def get_measurements_tensor(self, by):
+        """ Returns a tensor version of the measurements array, split by an 
+        observation descriptor
 
+        Args:
+            by(String):
+                the descriptor by which the splitting is made
+
+        Returns:
+            measurements_tensor (ndarray):
+                3d array with the selected obs_descriptor constituting the
+                third dimension.
+        """
+        assert len(self.obs_descriptors.keys()) > 1, \
+            "dataset has too few obs_descriptors to build a tensor from it"
+        assert by in self.obs_descriptors.keys(), \
+            "third dimension not in obs_descriptors"
+        unique_values = get_unique_unsorted(self.obs_descriptors[by])
+        measurements_list = []
+        for v in unique_values:
```

Comment:
I think this part expects that the data for each slice is in the same order. I would prefer a variant which ensures that the matching is not confused when we generate this.
One very thorough version of this would make the user specify all dimensions they want to have in their array and average over the rest.
As a minimal solution, tell the user that this just pastes measurements in the same order as they occur in the dataset.